From 93fde0fa56e791ec0b343a462756b24d3f6a9cbb Mon Sep 17 00:00:00 2001
From: Arthur Jiang <twjiang@nuvoton.com>
Date: Thu, 19 Sep 2024 19:01:46 -0600
Subject: [PATCH] package/pkg-generic.mk: a more generic way to rebuild package

---
 package/pkg-generic.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/package/pkg-generic.mk b/package/pkg-generic.mk
index 45589bcbb4..c6fccc0377 100644
--- a/package/pkg-generic.mk
+++ b/package/pkg-generic.mk
@@ -1004,6 +1004,10 @@ $(1)-clean-for-rebuild: $(1)-clean-for-reinstall
 			rm -f $$($(2)_TARGET_BUILD)
 
 $(1)-rebuild:		$(1)-clean-for-rebuild $(1)
+			@if [[ ! -z "$$($(2)_RDEPENDENCIES)" ]]; then \
+				$(call MESSAGE,"these packages "'"$$($(2)_RDEPENDENCIES)"'" depend on $(1)"); \
+				make $$(foreach p,$$($(2)_RDEPENDENCIES),$$(p)-rebuild); \
+			fi
 
 $(1)-clean-for-reconfigure: $(1)-clean-for-rebuild
 			rm -f $$($(2)_TARGET_CONFIGURE)
-- 
2.43.0

